### PR TITLE
Modifier extension function for Dashed Round Rect

### LINKED ISSUE #466: Add a dashedRoundRect modifier extension function to the design system module
I think we can have a shared function for this.

image

https://github.com/DroidKaigi/conference-app-2024/blob/0d6f6c4b2a8fe563172de41415a2952134278cb1/feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt#L79

https://github.com/DroidKaigi/conference-app-2024/blob/a14a7525f49107c46978164c74df108f2cdea4b1/feature/about/src/commonMain/kotlin/io/github/droidkaigi/confsched/about/component/AboutDroidKaigiDetailSummaryCard.kt#L46

### PR DESCRIPTION
**Pull request by ked4ma**
## Issue
- close #466 

## Overview (Required)
- some codes have similar implements to display "dashed round rect".
- this pr aim enabling to do it byproviding extension function 

## Links
-

### CODE REVIEW

File: feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt [hunk 1]
```diff
@@ -71,16 +68,12 @@ fun TimetableItemDetailSummaryCard(
...
-                    ),
-                    cornerRadius = CornerRadius(4.dp.toPx()),
-                )
-            }
+            .dashedRoundRect(
+                color = primaryColor,
+                cornerRadius = 4.dp,
+                strokeWidth = 1.dp,
+                intervals = arrayOf(2.dp, 2.dp),
+            )
```
Comment (ked4ma) [inline feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt:76]: To call `toPx()`, `DrawScope` or `Density` is needed.
(`drawBehind` receives `DrawScope.() -> Unit` as its param)

And passing float is a bit difficult to image how long it is.
So I defined this func with `Dp`s.

Of cource `2f` and `5f` are not equal to `1.dp`, `2.dp` but when I run the app, it was almost same view for me. (little bit bright though)

|before|after|
|--|--|
|||

File: feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt [hunk 2]
```diff
@@ -71,16 +68,12 @@ fun TimetableItemDetailSummaryCard(
...
-                        pathEffect = PathEffect.dashPathEffect(floatArrayOf(5f, 5f), 0f),
-                    ),
-                    cornerRadius = CornerRadius(4.dp.toPx()),
-                )
-            }
+            .dashedRoundRect(
+                color = primaryColor,
+                cornerRadius = 4.dp,
+                strokeWidth = 1.dp,
+                intervals = arrayOf(2.dp, 2.dp),
```
Comment (takahirom) [inline feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt:75]: Can we have these parameters as dashedRoundRect's default parameters?

Comment (ked4ma) [reply feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableItemDetailSummaryCard.kt]: update dashedRoundRect func :+1: 
https://github.com/DroidKaigi/conference-app-2024/pull/476/commits/d921847cdc9877f8a2676f9d892c5c841b0adc15

**Review by takahirom — APPROVED** [review]
Looks great! You nailed it!

**Pull request merged**